### PR TITLE
Pyspark3.2.3

### PR DESCRIPTION
**Pull request by bjornjorgensen**
## Describe your changes
Downgrade spark from 3.3.1 to 3.2.3
## Issue ticket if applicable
Fix: https://github.com/jupyter/docker-stacks/issues/1838

## Checklist (especially for first-time contributors)

- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [x] I will try not to use force-push to make the review process easier for reviewers
- [x] I have updated the documentation for significant changes

### CODE REVIEW

**Comment by mathbunnyru** [comment]
@bjornjorgensen please, take a look why this build fails.

**Comment by bjornjorgensen** [comment]
test local 

docker build --rm --force-rm -t pyspark-notebook:spark-3.2.3-def .
Sending build context to Docker daemon  7.168kB
Step 1/24 : ARG OWNER=jupyter
Step 2/24 : ARG BASE_CONTAINER=$OWNER/scipy-notebook
Step 3/24 : FROM $BASE_CONTAINER
 ---> 15d01b943cb7
Step 4/24 : LABEL maintainer="Jupyter Project <jupyter@googlegroups.com>"
 ---> Using cache
 ---> 6c2095ebbe2b
Step 5/24 : SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 ---> Using cache
 ---> b9292a47b1b6
Step 6/24 : USER root
 ---> Using cache
 ---> c0b3d44e6204
Step 7/24 : ARG spark_version="3.2.3"
 ---> Running in 03398999d6da
Removing intermediate container 03398999d6da
 ---> e599dcca0fba
Step 8/24 : ARG hadoop_version="3.2"
 ---> Running in dce3878ab10c
Removing intermediate container dce3878ab10c
 ---> 76c4019b6be9
Step 9/24 : ARG scala_version
 ---> Running in 5561f5a909be
Removing intermediate container 5561f5a909be
 ---> 2eb954c327ee
Step 10/24 : ARG spark_checksum="620c336a6b742c5c8da9c916b102d0795907e10f73b5ab7808f8f68b5e091766024bc9c28607806351177807d4c8b28b5e2de733aaad1f17e0676c466b520d7e"
 ---> Running in 5d88798d4aa2
Removing intermediate container 5d88798d4aa2
 ---> 1de0f5bd5e21
Step 11/24 : ARG openjdk_version="17"
 ---> Running in 34138f9f3680
Removing intermediate container 34138f9f3680
 ---> 30fe5ef7d68e
Step 12/24 : ENV APACHE_SPARK_VERSION="${spark_version}"     HADOOP_VERSION="${hadoop_version}"
 ---> Running in f3dc513623a0

 ...
 
 Step 15/24 : RUN if [ -z "${scala_version}" ]; then     wget -qO "spark.tgz" "https://archive.apache.org/dist/spark/spark-${APACHE_SPARK_VERSION}/spark-${APACHE_SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}.tgz";   else     wget -qO "spark.tgz" "https://archive.apache.org/dist/spark/spark-${APACHE_SPARK_VERSION}/spark-${APACHE_SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}-scala${scala_version}.tgz";   fi &&   echo "${spark_checksum} *spark.tgz" | sha512sum -c - &&   tar xzf "spark.tgz" -C /usr/local --owner root --group root --no-same-owner &&   rm "spark.tgz"
 ---> Running in 3de4b4cde6f8
spark.tgz: OK
Removing intermediate container 3de4b4cde6f8
 ---> 82876963e642
Step 16/24 : ENV SPARK_HOME=/usr/local/spark
 ---> Running in 28848992a7b9
Removing intermediate container 28848992a7b9
 ---> 0d407b800bbf
Step 17/24 : ENV SPARK_OPTS="--driver-java-options=-Xms1024M --driver-java-options=-Xmx4096M --driver-java-options=-Dlog4j.logLevel=info"     PATH="${PATH}:${SPARK_HOME}/bin"
 ---> Running in 75f2100205f0
Removing intermediate container 75f2100205f0
 ---> 36af654fab92
Step 18/24 : RUN if [ -z "${scala_version}" ]; then     ln -s "spark-${APACHE_SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}" "${SPARK_HOME}";   else     ln -s "spark-${APACHE_SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}-scala${scala_version}" "${SPARK_HOME}";   fi &&   mkdir -p /usr/local/bin/before-notebook.d &&   ln -s "${SPARK_HOME}/sbin/spark-config.sh" /usr/local/bin/before-notebook.d/spark-config.sh
 ---> Running in d68178765639

 ...
 
Successfully built 6646dec25aa5
Successfully tagged pyspark-notebook:spark-3.2.3-def

**Comment by bjornjorgensen** [comment]
FAILED tests/all-spark-notebook/test_spark_notebooks.py::test_nbconvert[local_sparkR] - assert not ['ERROR: Error in checkJavaVersion(): Java version, greater than or equal to 8 and less than 12, is required for this package; found version: 17.0.5']
I will change java to 11 then.

**Comment by bjornjorgensen** [comment]
@mathbunnyru where does platform linux -- Python 3.11.0, pytest-7.2.0, pluggy-1.0.0 this come from?

**Comment by bjornjorgensen** [comment]
One of this errors seams to be https://issues.apache.org/jira/browse/SPARK-27981
We can try with java 8 then.

**Comment by mathbunnyru** [comment]
> @mathbunnyru where does platform linux -- Python 3.11.0, pytest-7.2.0, pluggy-1.0.0 this come from?

I'm not 100% sure what you're referring to, but we install the latest python version in GitHub and then install pytest.
pluggy is probably a dependency of some package we install.

**Comment by bjornjorgensen** [comment]
@bsikander I did a local build with java 11 and it seams to work, but her we have to change to match to get it working. 
I can push the image that I have build with java 11 to my dockerhub repo if you want to. 
It then will be bjornjorgensen/sparksomething or you can build it your self.

**Comment by bjornjorgensen** [comment]
> > @mathbunnyru where does platform linux -- Python 3.11.0, pytest-7.2.0, pluggy-1.0.0 this come from?
> 
> I'm not 100% sure what you're referring to, but we install the latest python version in GitHub and then install pytest. pluggy is probably a dependency of some package we install.

It was python 3.11 I wos thinkig of..

**Comment by mathbunnyru** [comment]
> > > @mathbunnyru where does platform linux -- Python 3.11.0, pytest-7.2.0, pluggy-1.0.0 this come from?
> > 
> > 
> > I'm not 100% sure what you're referring to, but we install the latest python version in GitHub and then install pytest. pluggy is probably a dependency of some package we install.
> 
> It was python 3.11 I wos thinkig of..

It's not python version inside the image, but the version of the python in the runner.
For `aarch64`, it will be system python version, for `x86_64`, latest installed by this line:
https://github.com/jupyter/docker-stacks/blob/main/.github/actions/create-dev-env/action.yml#LL19

It doesn't affect what happens inside the image, so nothing to worry about.

File: pyspark-notebook/Dockerfile [hunk 1]
```diff
@@ -15,11 +15,11 @@ USER root
 # Spark dependencies
 # Default values can be overridden at build time
 # (ARGS are in lower case to distinguish them from ENV)
-ARG spark_version="3.3.1"
-ARG hadoop_version="3"
+ARG spark_version="3.2.3"
+ARG hadoop_version="3.2"
 ARG scala_version
```
Comment (consideRatio) [inline pyspark-notebook/Dockerfile:20]: A comment to clarify when a newer version can be adopted would be helpful in situations like these.

Comment (bjornjorgensen) [reply pyspark-notebook/Dockerfile:20]: This is a special case. I have backport a PR to branch 3.2. 
The hope here wos to have a build with spark 3.2.3 and then revers this PR back to 3.3.1

**Comment by bsikander** [comment]
@bjornjorgensen it would be great if you can push the image to your docker hub. Thanks.

**Comment by bjornjorgensen** [comment]
[bjornjorgensen/pyspark-notebook:spark-3.2.3-bj](https://hub.docker.com/repository/docker/bjornjorgensen/pyspark-notebook)

**Pull request closed without merging**